dags/pull_counts.py

```diff
@@ -0,0 +1,90 @@
+#!/data/airflow/airflow_venv/bin/python3
+# -*- coding: utf-8 -*-
+# noqa: D415
+r"""### The Daily counts Replicator DAG
+
+This DAG runs daily to update the following counts tables from the tables
+staged in the database by the MOVE's ``bigdata_replicator`` DAG:
+
+1\. arteries_centreline
+
+2\. arteries_counts_groups
+
+3\. arteries_groups
+
+4\. arterydata
+
+5\. atr_metadata
+
+6\. atr_study_data
+
+7\. cnt_det
+
+8\. countinfo
+
+9\. counts_multiday_runs
+
+10\. oracle_cnt_det
+
+11\. studies
+"""
+import os
+import sys
+from datetime import timedelta
+from functools import partial
+import pendulum
+# pylint: disable=import-error
+from airflow.decorators import dag
+from airflow.models import Variable
+
+# import custom operators and helper functions
+repo_path = os.path.abspath(os.path.dirname(os.path.dirname(os.path.realpath(__file__))))
+sys.path.insert(0, repo_path)
+# pylint: disable=wrong-import-position
+from dags.dag_functions import task_fail_slack_alert
+# pylint: enable=import-error
+
+DAG_NAME = "counts_replicator"
+DAG_OWNERS = Variable.get("dag_owners", deserialize_json=True).get(DAG_NAME, ["Unknown"])
+
+default_args = {
+    "owner": ",".join(DAG_OWNERS),
+    "depends_on_past": False,
+    "start_date": pendulum.datetime(2023, 10, 31, tz="America/Toronto"),
+    "email_on_failure": False,
+    "retries": 3,
+    "retry_delay": timedelta(minutes=60),
+    "on_failure_callback": task_fail_slack_alert,
+}
+
+@dag(
+    dag_id=DAG_NAME,
+    default_args=default_args,
+    catchup=False,
+    max_active_runs=5,
+    schedule_interval=None,
+    doc_md=__doc__,
+    tags=["counts"]
+)
+def counts_replicator():
+    """The main function of the counts DAG."""
+    from dags.common_tasks import wait_for_external_trigger, copy_table
+
+    tables = [
```

Comment:
I do wonder if dynamic task mapping is the best choice for this when the tables to be replicated are hard coded in the DAG. In this case it may be more clear to generate them with a for loop (the old dynamic way), perhaps adding start/end dates to the list. If the tables are changing every day, then it would make more sense to use dynamic task mapping and pass the tables to be replicated with the xcom value from the external trigger. 
Obviously this approach is working fine, but curious to hear your thoughts!

Reply:
you're right. As per Airflow documentation, this's the simplest dynamic task mapping. I was going to read the table names from an Airflow variable to avoid hard-coding these names.
I also second your proposal to get the list of tables to be copied from the external trigger. But I considered this as a future work as it requires some more work to send this list of names using Airflow APIs between the two DAGs and to create these tables if they don't already exist. To keep this PR simpler and get the DAGs ready sooner, I'll create a new issue for this *"improvements"* to be done in the future.